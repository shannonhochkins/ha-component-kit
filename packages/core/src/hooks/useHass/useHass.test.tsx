import {
  TestWrapper,
  onReady,
  mocked,
  connection,
} from "@mocks/mockConnection";
import { renderHook, act, waitFor } from "@testing-library/react";
import { useHass } from "@core";

describe("useHass", () => {
  beforeEach(() => {
    onReady.mockClear();
  });
  describe("callService", () => {
    it("should allow a user to call a service with camel case values", async () => {
      const { result } = renderHook(() => useHass(), {
        wrapper: TestWrapper,
      });
      await waitFor(() => expect(onReady).toHaveBeenCalledTimes(1));
      act(() => {
        result.current.callService({
          domain: "mediaPlayer",
          service: "volumeSet",
          serviceData: {
            volume_level: 10,
          },
          target: "media_player.fake_player",
        });
      });

      expect(mocked.callService).toHaveBeenCalledWith(
        connection,
        "media_player",
        "volume_set",
        {
          volume_level: 10,
        },
        {
          entity_id: "media_player.fake_player",
        }
      );
    });

    it("should allow a user to call a service with snake case values", async () => {
      const { result } = renderHook(() => useHass(), {
        wrapper: TestWrapper,
      });
      await waitFor(() => expect(onReady).toHaveBeenCalledTimes(1));
      act(() => {
        result.current.callService({
          domain: "media_player",
          service: "volume_set",
          serviceData: {
            volume_level: 10,
          },
          target: "media_player.fake_player",
        });
      });

      expect(mocked.callService).toHaveBeenCalledWith(
        connection,
        "media_player",
        "volume_set",
        {
          volume_level: 10,
        },
        {
          entity_id: "media_player.fake_player",
        }
      );
    });
  });

  describe("getAllEntities", () => {
    it("should retrieve all the entities", async () => {
      const { result } = renderHook(() => useHass(), {
        wrapper: TestWrapper,
      });
      await waitFor(() => expect(onReady).toHaveBeenCalledTimes(1));
      const entities = result.current.getAllEntities();
      expect(entities).toMatchSnapshot();
    });
  });
});
