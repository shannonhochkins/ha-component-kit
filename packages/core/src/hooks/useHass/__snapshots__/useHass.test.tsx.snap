// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`useHass getAllEntities should retrieve all the entities 1`] = `"{"light.fake_light":{"attributes":{"friendly_name":"Dining room light","icon":"mdi:light-recessed","min_color_temp_kelvin":2702,"max_color_temp_kelvin":6535,"min_mireds":153,"max_mireds":370,"effect_list":["Night","Read","Meeting","Leasure","Soft","Rainbow","Shine","Beautiful","Music"],"supported_color_modes":["color_temp","hs"],"color_mode":"hs","brightness":255,"hs_color":[9,67.1],"rgb_color":[255,109,83],"xy_color":[0.591,0.328],"raw_state":true,"supported_features":23},"state":"on","entity_id":"light.fake_light","last_changed":"<DATE_PLACEHOLDER>","last_updated":"<DATE_PLACEHOLDER>","context":{"id":"","user_id":null,"parent_id":null}},"switch.fake_gaming_switch":{"attributes":{"friendly_name":"Gaming Computer"},"state":"off","entity_id":"switch.fake_gaming_switch","last_changed":"<DATE_PLACEHOLDER>","last_updated":"<DATE_PLACEHOLDER>","context":{"id":"","user_id":null,"parent_id":null}},"media_player.fake_tv":{"attributes":{"friendly_name":"Living room TV"},"state":"off","entity_id":"media_player.fake_tv","last_changed":"<DATE_PLACEHOLDER>","last_updated":"<DATE_PLACEHOLDER>","context":{"id":"","user_id":null,"parent_id":null}},"scene.good_morning":{"entity_id":"scene.goodmorning","state":"2023-07-11T04:36:54.685Z","attributes":{"entity_id":["light.all_office_lights_2"],"id":"1688447567040","icon":"mdi:sun-clock","friendly_name":"Goodmorning"},"context":{"id":"01H4FP0HY3K7R49GJS574K5TPW","parent_id":null,"user_id":null},"last_changed":"<DATE_PLACEHOLDER>","last_updated":"<DATE_PLACEHOLDER>"},"sensor.time":{"entity_id":"sensor.time","state":"14:36","attributes":{"icon":"mdi:clock","friendly_name":"Time"},"context":{"id":"01H4JAXGF1RTA2MJGGPGAGM7VD","parent_id":null,"user_id":null},"last_changed":"<DATE_PLACEHOLDER>","last_updated":"<DATE_PLACEHOLDER>"},"climate.air_conditioner":{"entity_id":"climate.air_conditioner","state":"off","attributes":{"hvac_modes":["fan_only","dry","cool","heat","heat_cool","off"],"min_temp":7,"max_temp":35,"target_temp_step":1,"fan_modes":["Low","Mid","High"],"current_temperature":24,"temperature":25,"fan_mode":"High","hvac_action":"off","friendly_name":"Air Conditioner","supported_features":9},"context":{"id":"01H4J3SQV4JJX4KF6G28K2AADY","parent_id":null,"user_id":null},"last_changed":"<DATE_PLACEHOLDER>","last_updated":"<DATE_PLACEHOLDER>"}}"`;
